### DISCUSSION TITLE
Update an OSXPhotos Backup from a new Photo Library

### DISCUSSION THREAD

**Discussion by jeremieberrebi** [RhetTbull/osxphotos#1415] [Q&A]
Hello @RhetTbull 

We discussed in this thread https://github.com/RhetTbull/osxphotos/discussions/1400

Finally, I reimported all my photos from the different folders to a new Photo Library. All is working perfectly now. 
(I had a few hundreds pictures missing but they were not in my original Photo Library (perhaps, former deleted files?) but it's not the subject of my question)

What I've done:
Photo Library 1 >>> OsXPhotos Backup >>> New Photo Library

I want to always keep a backup of files made with OSXPhotos. I think it's always great to have the original pictures as backup.

I want this:
Photo Library 1 >>> OsXPhotos Backup >>> New Photo Library >>> OsXPhotos Backup updated regularly.

Now that I have my new Photo Library, do I have to reexport everything once before to use your --update parameter? Or do you think I could avoid to reexport and just add new pictures to the former export?

By reading your documentation, it seems that I have to reexport everything...to have a new export database.

Thanks again

**Comment by RhetTbull** [comment]
Hi @jeremieberrebi Yes, you can do this with the `--update` command to perform an incremental backup to the old backup. BUT first you must migrate the backup to the new library using this command:

`osxphotos exportdb /path/to/OSXPhotosBackup --migrate-photos-library /path/to/NewPhotoLibrary --verbose`

This step is required because osxphotos (like Photos) keeps track of photos and videos using a unique identifier known as a UUID. When you created the new library the photos would each get a new UUID and so osxphotos wouldn't recognize them as the old photos and would re-export. The migration command will match the photos in the new library to those in the old library and update the export database to work with the new library. You only need to do this one time.

**Comment by jeremieberrebi** [comment]
Thanks a lot @RhetTbull ! Impressive!
I launched the script already since yesterday but it seems it will take a long time. I have 140000 assets in the Library and it seems that it's taking around 10 seconds per asset to recognize it.
If I'm not wrong it will take 16 days to finalize this. 
Perhaps you have an idea on how can I speed it up?

The OSXPhotos backup is on a SSD Drive on a Synology NAS
The Library is on an External Hard Drive connected to an iMac M3 with 16GB Ram.

If not, perhaps, it will be better to reexport everything from the new Library

**Reply by RhetTbull** [reply]
> The OSXPhotos backup is on a SSD Drive on a Synology NAS

This is likely the problem. Synology NAS are **VERY** slow when working with SQLite databases which is what OSXPhotos uses to store the export state in. That's the reason for the `--ramdb` option with `osxphotos export` but unfortunately the migrate option doesn't support `--ramdb`.  The update requires a read from the database then 2 executes and 2 commits and each of those is probably very slow given your export database (`.osxphotos_export.db` in the export folder) is likely large (~1GB) given the size of your library. This might work:

Cancel the migration.

osxphotos will have created a backup library on the NAS named something like: `.osxphotos_export.db.20240227090306.bak`  There may be up to three files associated with this -- the `.db`, `.db-wal`, and `.db-shm`.  Copy all three to a new folder on the Mac's internal disk. Rename them to `.osxphotos_export.db`, `.osxphotos_export.db-shm`, and `.osxphotos_export.db-wal` respectively.  Re-run the migration command on this new folder then when it's done, copy the datbase file(s) back to the original export location.

I could work on adding a `--ramdb` option to migrate but I don't use a Synology and this is an edge case so it's pretty low on my priority list.  I'll add an issue to track it though.

**Reply by RhetTbull** [reply]
I've also opened an issue to auto-detect if Synology being used and if so, use the RAM DB option. Synology is just far too slow when used with SQLite which requires lots of reads/writes to a file.

**Reply by jeremieberrebi** [reply]
OK! I cancelled the migration @RhetTbull 

I moved the files to my Downloads Folder on my internal disk and renamed them
In Terminal, I went to the Downloads folder
I typed:
jeremieberrebi@Jeremies-iMac Downloads % **osxphotos exportdb /Volumes/Photos\ SBackup --migrate-photos-library /Volumes/SEAGATE\ 4TO/New\ Photo\ Library.photoslibrary --verbose**

and I got the messages at the bottom of this message. 
It seems that the command is not using at all the files who are in the Downloads folder and recreated export database on the Synology : /Volumes/Photos\ SBackup
**How can I tell the command that the database it needs to use is the one in the Downloads folder?**
It seems that launching the command from the Downloads folder is not sufficient.


⚠️  This command will update your export database (/Volumes/Photos SBackup/.osxphotos_export.db) 
to use /Volumes/SEAGATE 4TO/New Photo Library.photoslibrary as the new source library.
The last library used was /Volumes/SEAGATE 4TO/Photos Library.photoslibrary.
This will allow you to use the export database with the new library but it will
no longer work correctly with the old library unless you run the `--migrate-photos-library`
command again to update the export database to use the previous library.

A backup of the export database will be created in the same directory as the export database.

Do you want to continue? [y/N]: y
Backed up export database to /Volumes/Photos SBackup/.osxphotos_export.db.20240227192148.bak
Loading data from export database /Volumes/Photos SBackup/.osxphotos_export.db
Loading data from Photos library /Volumes/SEAGATE 4TO/New Photo Library.photoslibrary
2024-02-27 19:25:20,657 - root - WARNING - photosdb.py - 157 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 12.0, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 13.0, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 14.0, 14.1]: you have Darwin, OS version: 14.3
Processing database /Volumes/SEAGATE 4TO/New Photo Library.photoslibrary/database/photos.db
Processing database /Volumes/SEAGATE 4TO/New Photo Library.photoslibrary/database/Photos.sqlite
Processing database.
Database version: 6000, 9.
Processing persons in photos.
Processing detected faces in photos.
Processing albums.
Processing keywords.
Processing photo details.

**Reply by RhetTbull** [reply]
OK, I didn't realize it was reading the export path from the database itself. This may require a code fix whichI likely won't be able to look at until the weekend at earliest. Your best bet might be to re-export everything. I'll look at a fix but it may take some time.

**Reply by jeremieberrebi** [reply]
Yep. I'll reexport everything :)
Thanks for everything!
Is there a way I can donate something for your help?

**Reply by RhetTbull** [reply]
> In Terminal, I went to the Downloads folder
> I typed:
> jeremieberrebi@Jeremies-iMac Downloads % osxphotos exportdb /Volumes/Photos\ SBackup --migrate-photos-library > /Volumes/SEAGATE\ 4TO/New\ Photo\ Library.photoslibrary --verbose


I just re-read your command and realized what you did wrong. With the `osxphotos exportdb` command you need to specify the path to where the export database is located, either the parent directory or the actual path to the database. You specified the path to the old database, not the new database you copied. You should have run:

`osxphotos exportdb ~/Downloads --migrate-photos-library  /Volumes/SEAGATE\ 4TO/New\ Photo\ Library.photoslibrary --verbose`

Or:

`osxphotos exportdb ~/Downloads/.osxphotos_export.db --migrate-photos-library  /Volumes/SEAGATE\ 4TO/New\ Photo\ Library.photoslibrary --verbose`

I tested this and it works as expected.

**Reply by jeremieberrebi** [reply]
Ouch, too late. I've erased everything and launched a new backup. In 7 hours, it exported 11% of the database. It shows that there are 15 hours left to finish everything but that doesn't seem logical to me. There are 147069 photos/videos to export.

Regarding what you wrote to me, that's exactly what I was thinking. I was sure there was a way to specify the location of the database I moved to Downloads, and that's what I was asking you about. However, I thought it was also mandatory to specify where the photos are located, which is why I left the backup path, but I understand now that it wasn't necessary and that the database alone can update even without access to the photos, which is good to know for the future.

Too bad but in the other hand, perhaps better to have a complete new export from scratch.
Thank you very much.

What will be the command I will need to use in the future to regularly update the backup? Won't I have a problem again if the database is located on the Synology?

The command I used for the backup this time is:
osxphotos export --skip-edited --retry 3 /volumes/LibraryBackup --ramdb --export-by-date

**Reply by RhetTbull** [reply]
> What will be the command I will need to use in the future to regularly update the backup? Won't I have a problem again if the database is located on the Synology?

> The command I used for the backup this time is: osxphotos export --skip-edited --retry 3 /volumes/LibraryBackup --ramdb --export-by-date

Use the same command but add `--update`

**Reply by jeremieberrebi** [reply]
New export went well! 

Exporting 147069 photos to /volumes/LibraryBackup...
Exporting 147069 photos ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Processed: 147069 photos, exported: 154303, missing: 0, error: 0
Elapsed time: 1 day, 20:48:19

If I delete a picture in the Photo Library, what will happen if I use the --update parameter?

**Reply by RhetTbull** [reply]
When you run `--update` this is what happens:

- Any photos/videos that have been added to the library since the last export will be exported
- Any photo that has changed (e.g. been edited) since the last export will be re-exported and updated (the exported copy replaced with the new version). If you use `--exiftool` or `--sidecar` and the metadata for the photo changed (for example, you added a keyword), the appropriate action will be taken to ensure the photo or the sidecar contains the new metadata.
- Any photo deleted from the library will be ignored. The previously exported photo will be left in the export and will be ignored by osxphotos.

If instead, you want photos deleted from Photos library to also be deleted from the backup, add the `--cleanup` flag: `--update --cleanup`
 
This will cause any asset that was deleted from the library to also be deleted from the export folder. Any other files in the export folder that "don't belong" to the backup will also be deleted. For example, if you manually added a file to the backup folder, it would also be deleted because OSXPhotos would not know about it.  See the `--keep` option if you need `--cleanup` to keep certain files: `osxphotos help export keep`

**Reply by jeremieberrebi** [reply]
Hello @RhetTbull 

I had to rebuild my Photo Library using the Rebuild function in Photos. Many items disappeared, such as the faces. I want to continue updating my OSXPhotos backup. Can I still use the --update feature, or did the rebuild reset everything?

**Reply by RhetTbull** [reply]
@jeremieberrebi It is possible the repair Photos library function changed all the unique IDs associated with each photo. This does not always happen but I've seen Photos do this. To be safe,  I recommend the migrate feature discussed above:

`osxphotos exportdb /path/to/OSXPhotosBackup --migrate-photos-library /path/to/NewPhotoLibrary --verbose`

In this case, the ` /path/to/NewPhotoLibrary ` will be the same as the existing library but this will force osxphotos to match each photo ID to the correct ID in the database.

**Reply by jeremieberrebi** [reply]
I'm afraid to have the same problem than the one I had here. It could be very slow. I have 153000 photos and 12000 videos.
https://github.com/RhetTbull/osxphotos/discussions/1415#discussioncomment-8608181

My OSXPhotosBackup is on a Synology NAS....

**Reply by RhetTbull** [reply]
I'd forgotten about that. I opened a new issue to fix this -- will get a fix done as soon as I have time. In the meantime, you could dry an export with `--update and --dry-run` and see if it works. With `--dry-run` nothing is exported. If you see a lot of photos being re-exported, you'll know the IDs changed and we need to update the export database. If not, then you're probably good to go.

**Reply by jeremieberrebi** [reply]
Just tried
It's reimporting everything

Perhaps I could use the method you described here?
https://github.com/RhetTbull/osxphotos/discussions/1415#discussioncomment-8611760


Something else
It's taking a long time to begin the export because it's processing Metadatas, people, albums etc... (see below)
In my export I'm not exporting anything regarding that. Is it possible to bypass that?

jeremieberrebi@Jeremies-iMac ~ % osxphotos export --skip-edited --retry 3  /volumes/LibraryBackup --ramdb --export-by-date --update --dry-run
Using last opened Photos library: /Volumes/SEAGATE 4TO/New Photo Library.photoslibrary
2024-03-20 06:45:46,943 - root - WARNING - photosdb.py - 157 - WARNING: This module has only been tested with macOS versions [10.12, 10.13, 10.14, 10.15, 10.16, 11.0, 11.1, 11.2, 11.3, 11.4, 11.5, 11.6, 11.7, 12.0, 12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7, 13.0, 13.1, 13.2, 13.3, 13.4, 13.5, 13.6, 14.0, 14.1]: you have Darwin, OS version: 14.4
Processing database /Volumes/SEAGATE 4TO/New Photo 
Library.photoslibrary/database/photos.db
Processing database /Volumes/SEAGATE 4TO/New Photo 
Library.photoslibrary/database/Photos.sqlite
Processing database.
Database version: 6000, 9.
Processing persons in photos.
Processing detected faces in photos.
Processing albums.
Processing keywords.
Processing photo details.
Processing import sessions.
Processing additional photo details.
Processing face details.
Processing photo labels.
Processing EXIF details.
Processing computed aesthetic scores.
Processing comments and likes for shared photos.
Processing moments.
Processing syndication info.
Processing shared iCloud library info
Done processing details from Photos library.
Exporting 165427 photos to /volumes/LibraryBackup...
Exporting 165427 photos ━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━  39% 0:26

**Reply by RhetTbull** [reply]
> Perhaps I could use the method you described here?
> https://github.com/RhetTbull/osxphotos/discussions/1415#discussioncomment-8611760

The problem is your NAS setup will make this very slow and unfortunately osxphotos doesn't support moving the export database to a temp location until I fix the issue (#1450) with this you'll have to update the DB in place over the NAS and your experience shows this to be very slow.

> It's taking a long time to begin the export because it's processing Metadatas, people, albums etc... (see below)
> In my export I'm not exporting anything regarding that. Is it possible to bypass that?

No. This is a core part of how osxphotos works. It must read in the entire Photos library before it can begin export. Someday I might fix this but it requires a complete rewrite of much of the osxphotos code. "osxphotos is too slow" is not something I spend much time on because export is something you can start and walk away until it's done.

**Comment by rajscode** [comment]
@jeremieberrebi unrelated to OSXPhotos app itself, but since you are dealing with a lot of files and some of them relatively small, I was wondering if your mac <---> nas is optimized to possibly get some more performance out of it ... 

Couple questions and suggustions potentially, especially if SMB is being used. 
- What MacOS version are you running
- Synology model ? What version of DSM ?
- Number of disks and type (HD or SSD ?)
- RAID type
- Volume Type btrfs or ext4. Is network logging or compression enabled on the Synology volume 
- What file service(s) are enabled on the NAS. If more then 1, which one are are you using between the Mac and NAS to access the NAS shared folder (SMB, NFS, AFP, FTP, etc.) 
- if SMB, on the Synology NAS inside control panel -- File Services, SMB advanced settings, what are all the settings currently configured as?
- if SMB do you have a /etc/nsmb.conf file on your Mac(s).  What are the contents of it ? 
- Your network environement locally is ? (Ethernet or 1 gigabit or 2.5 gigabit, 10GbE etc..)
- Your mac and NAS are connected via a switch / router or direct connect ? 
- What connection type - ethernet or 1 gigabit or 2.5G, etc.. 
- Are you using jumbo frames instead of the standard 1500 byte MTU for both the NAS and the Mac (as well as any other devices access the NAS)
- Are both the wired NIC and Wifi enabled at same time on your Mac ? If so what is the Network Connection Service Order set as (It should have your wired NIC as the highest position imo)

re: SMB only, the following config file settings on the Mac for /etc/nsmb.conf may help speed up your transfers. This will disable packet signing, caching and indexing while browsing SMB shares. Forces SMB 3, enables multichannel connections, and prioritizes wired connections over wireless.  Will need to reboot your Mac for this to take effect.  Recommend you backup existing nsmb.conf file before replacing it's contents with below (so you can revert back if needed).

/etc/nsmb.conf file contents:
```shell
[default]
signing_required=no
streams=yes
notify_off=yes
port445=no_netbios
soft=yes
dir_cache_max_cnt=0
dir_cache_max=0
dir_cache_off=yes
protocol_vers_map=4
validate_neg_off=yes
mc_on=yes
mc_prefer_wired=yes
```

**Reply by jeremieberrebi** [reply]
Hello @rajscode 

Sorry I was overloaded last week.

My answers below

Couple questions and suggestions potentially, especially if SMB is being used.

> What MacOS version are you running

I have an iMac M3 with 16GB of RAM
MacOS Sonoma 14.4.1 (23E224)

> Synology model ? What version of DSM ?

DS1522+
https://www.synology.com/en-us/products/DS1522+

DSM updated to the last version
7.2.1-69057 Update 4

> Number of disks and type (HD or SSD ?)

<img width="626" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/08e6e0f7-5b19-4fbf-85dd-6cffc8eb5975">

I'm using the SSD for the OXPhotos Library Backup

> RAID type

The first pool with the 2 first drive are in Raid 1
The second pool with the SSD is SHR (Synology Hybrid RAID)
The third pool with the 8TB Drive is also with SHR

> Volume Type btrfs or ext4. Is network logging or compression enabled on the Synology volume

First pool : ext4
SecondPool : Btrfs (The SSD with the OsxPhotos Library is here)
ThirdPool: Btrfs

> What file service(s) are enabled on the NAS. If more then 1, which one are are you using between the Mac and NAS to access the NAS shared folder (SMB, NFS, AFP, FTP, etc.)

SMB is activated
NFS is activated (Perhaps I need to deactivate it?)
Bonjour Service discovery is activated

I'm using SMB.


> if SMB, on the Synology NAS inside control panel -- File Services, SMB advanced settings, what are all the settings currently configured as?

<img width="594" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/92691078-8069-4c7d-b11f-bc23fdb32cbf">
<img width="593" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/4f6d0b31-ffa5-41c5-b10e-a0ced1dc57ca">
<img width="581" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/41def0ac-16a1-4f76-963f-efd1ed7fd2de">
<img width="586" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/0adf6c53-a249-45f2-9050-b0d74033d612">

> if SMB do you have a /etc/nsmb.conf file on your Mac(s). What are the contents of it ?

No file like this one

> Your network environement locally is ? (Ethernet or 1 gigabit or 2.5 gigabit, 10GbE etc..)

1000BaseT with CAT8 Cables.

> Your mac and NAS are connected via a switch / router or direct connect ?

via a TP-Link Router  [ARCHER AX53](https://www.tp-link.com/my/home-networking/wifi-router/archer-ax53/)

> What connection type - ethernet or 1 gigabit or 2.5G, etc..

Ethernet 1000BaseT with CAT8 Cables

> Are you using jumbo frames instead of the standard 1500 byte MTU for both the NAS and the Mac (as well as any other devices access the NAS)

For the moment, the Network interface appearing on the Synology is this one
1000 Mbps, Full duplex, MTU 1500

Same on the Mac

I never tried to use Jumbo frames.

> Are both the wired NIC and Wifi enabled at same time on your Mac ? If so what is the Network Connection Service Order set as (It should have your wired NIC as the highest position imo)

Only Wired is enable on the Mac.
Wifi is enabled on the router for another Mac (and other devices).


> re: SMB only, the following config file settings on the Mac for /etc/nsmb.conf may help speed up your transfers. This will disable packet signing, caching and indexing while browsing SMB shares. Forces SMB 3, enables multichannel connections, and prioritizes wired connections over wireless. Will need to reboot your Mac for this to take effect. Recommend you backup existing nsmb.conf file before replacing it's contents with below (so you can revert back if needed).

Are you telling me to create the file and add this? How will I know if I'll have better performance?

**Reply by jeremieberrebi** [reply]
By the way, it seems I can't use Jumbo Frames on my Synology DS1522+ without an hardware extension
https://www.reddit.com/r/synology/comments/12wed1l/synology_ds1522_jumbo_frames/

**Reply by jeremieberrebi** [reply]
@RhetTbull @rajscode 
I'm thinking about reexporting my entire database again instead of updating OSXPhotos.
It will take 2 days but at least, everything will be updated.
If you see something that I need to do better before to export, tell me.

For the moment, I'm using this command:
`osxphotos export --skip-edited --retry 3 /volumes/LibraryBackup --ramdb --export-by-date --update --dry-run`

I'm also thinking about adding a few other parameters to save more than just the pictures. Same, any advice will be appreciated.

**Reply by RhetTbull** [reply]
@jeremieberrebi  I think the command looks fine.

> I'm also thinking about adding a few other parameters to save more than just the pictures. Same, any advice will be appreciated.

I would add `--export-aae --sidecar XMP --report PhotosBackup_{today}.csv`

- `--export-aae`: exports the adjustments made to each edited photo in a way they can later be re-imported to Photos while preserving non-destructive edits if desired.
- `--sidecar XMP`: creates a metadata sidecar file in XMP format for each photo that preserves all the metadata such as keywords, etc.
- `--report PhotoBackup_{today}.csv`: creates a report file in CSV format with today's date of every action taken which is useful for looking at what had errors, etc.  

If you use albums to organize photos, these will be lost in this export. One thing you could do is add `--album-keyword` which will create a new keyword for each album name that is preserved in the XMP sidecar file.

**Reply by jeremieberrebi** [reply]
@RhetTbull Thanks again!

Is it possible to also keep the People present in a picture as keywords?

**Reply by jeremieberrebi** [reply]
It's ok I found the answer : --person-keyword

Let's go :)

**Reply by rajscode** [reply]
@jeremieberrebi 
- Don't worry about Jumbo Frames. Since your NAS unit model doesn't support it not an issue. 
- To establish a baseline idea of existing performance between your iMac M3 and the Synology NAS unit as currently configured, download a free app via the Mac App Store.  Blackmagic Disk Speed Test ( https://apps.apple.com/us/app/blackmagic-disk-speed-test/id425264550?mt=12 )
- Use the app above to establish a baseline of existing performance for read and write by pointing it at the connected volume / drive to be tested by changing the settings in the Speed Test App (setting icon right above the start button, select the target drive).  Basically once you've selected the target NAS volume hit the start button.  Let it get one measurement of MB/s for WRITE and one for READ.  Hit the start button again to STOP the test.  Write down those two numbers and note them as Baseline SMB1 values. 

**Changes # 1:**
Make the following changes on the Synology Unit (take screen shots / pictures of your original settings before making any changes so you can revert back if needed)
Synology Control Panel:
File Services
SMB Tab
Enable Transfer Log - OFF (recommend it be turned off)
SMB - Advanced Settings Button
General Tab
Maximum SMB Protocol: SMB3 
Minimum SMB Protocol: SMB2
Transport Encrypt Mode - Disable (As long as you aren't using Windows Active Directorey Domain capabilities on Synology )
Server Signing - Disable
Opportunistic Locking - Enable
Enable SMB2 Lease - Enable
Enable SMB Durable Handoles - Enable  (As long as AFP isn't being used on the Synology [tab next to the SMB tab on File Services ])

macOS Tab
Enable VFS - Check on
Uncheck cross-protocol with AFP

Others Tab
Enable SMB3 Multichannel

FileServices --> Advanced tab (far right after SMB / AFP / NFS / FTP ... / Advanced)
 - File Fast Clone - Check it On

Run the Black Magic Disk Test again. Make sure the target drive is properly selected. Run the test and record the new READ and WRITE numbers and label as: NAS UPDATED SMB2/3 Results. 

**Changes # 2:** 
On your iMac create the /etc/nsmb.conf file as per my original post. Reboot your iMac. Run the Black Magic Disk Test again. Make sure the target drive is properly selected. Run the test and record the new READ and WRITE numbers and label as: nssmb & SMB2/3 Result.  

Basically out of your three recorded results you want the one with the highest overall WRITE and READ MB/s results. If needed revert back via deleting the /etc/nsmb.conf file and rebooting the iMac.  And if needed to revert back to original undo any changes made on the NAS in CHANGES # 1 to what you had originally. Feel free to share your test results in all three configurations as it might be helpful for someone else if they are looking to see if it makes any difference:

i.e. 

| Config | Write | Read |
|--------|--------|--------|
| No Changes: Baseline SMB1 | ? | ? |
| Config Changes on NAS only: SMB2/3 | ? | ? |
| Config Changes on iMac (nssmb file) and NAS (SMB2/3) | ? | ? |

**Reply by jeremieberrebi** [reply]
@rajscode 
For the moment, I made the Changes #1. The speed doubled as you can see! Awesome!

<img width="680" alt="Blackmagic_Design_Disk_Speed_Test" src="https://github.com/RhetTbull/osxphotos/assets/4217084/4dadbfd5-b2dd-4934-bc3c-3e4ab5515cd6">
<img width="655" alt="Blackmagic_Design_Disk_Speed_Test" src="https://github.com/RhetTbull/osxphotos/assets/4217084/bb22fffe-059d-4c0b-95b9-7bbd19350d1e">

I'll work on Changes #2 tomorrow.

**Reply by jeremieberrebi** [reply]
@rajscode 
I made the Changes #2 but I got exactly the same results than after the Changes #1
How can I be sure that the Mac is really reading the nsmb.conf file?

<img width="667" alt="MacGPT_Sidebar" src="https://github.com/RhetTbull/osxphotos/assets/4217084/19fc61cd-9f14-4933-b515-07ef8cb88f45">

**Reply by rajscode** [reply]
> @rajscode 
> 
> How can I be sure that the Mac is really reading the nsmb.conf file?
> 
> 

Confirm that the nsmb.conf file is inside of the /etc directory.  And also that you rebooted the Mac once after the file was kept / created inside the /etc directory. 

If you did that then all good.

The expected biggest perf gain was from making the move off SMBv1.

Enjoy !
